.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,52 @@
+---
+name: Integration tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  integration-tests:
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        versions:
+          - ansible: stable-2.16
+            python: "3.10"
```

Comment:
Can we use a more recent Python version (e.g. 3.13)?
Are there any limitations from Ansible regarding this?

Reply:
Based on the ansible core version we are (currently) slated to support (2.16) it is compatible with python 3.10-3.12, I believe ansible core supports the 3 latest python versions at time of release.  Eventually as part of the release preparation for Galaxy/Automation Hub we'll want to modify the existing CI tests to run against a matrix of these different python versions.
https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-core-control-node-python-support

Looking at that link though I wonder if we should look to upgrade our "default" ansible core version to 2.17 or 2.18 as it buys us much more time before EoL and support for newer python versions.